xonsh/jobs.py

```diff
@@ -135,9 +135,19 @@ def give_terminal_to(pgid):
         def give_terminal_to(pgid):
             oldmask = signal.pthread_sigmask(signal.SIG_BLOCK,
                                              _block_when_giving)
-            os.tcsetpgrp(FD_STDERR, pgid)
-            signal.pthread_sigmask(signal.SIG_SETMASK, oldmask)
-            return True
+            try:
+                os.tcsetpgrp(FD_STDERR, pgid)
+                return True
+            except OSError as e:
+                if getattr(e, 'errno') == 22:  # [Errno 22] Invalid argument
```

Comment:
I think the getattr() call should either have a default value or it should be `e.errno`.  The current method doesn't add any safety and OSErrors should always have the `errno` attribute.